Os/Task.hpp

```diff
@@ -48,6 +48,7 @@ namespace Os {
              * Returns the task-handle owned by this task
              */
             POINTER_CAST getRawHandle();
+            void execRoutine(); // !< used by scheduler to start executing task. Do not explicitly call.
```

Comment:
I don't think you should define this function here.  Since it is a member function this has 2 issues:

1. Every creator of Os/Task has to define it or there will be undefined symbols.  This is the case even if you don't need this.
2. As you have pointed out, it would be dangerous to call this directly

A cleaner approach (I think) would be to put in a a Functor just in Posix/Task.cpp that might look  like this:

```c++
struct Functor {
     taskRoutine routine; //!< contains the task entrypoint
      void* arg; //!< contains the task entrypoint pointer
    
     void execute() {
          this->routine(this->arg);
    }
}
void* pthread_entry_wrapper(void* arg) {
     FW_ASSERT(arg);
     Functor* executor = (Functor*) arg;
     executor->execute();
     return NULL;
}
```
This could be created on the stack of the start call and doesn't add a bunch of members to Os::Task that other implementations won't use.

Reply:
I thought about this approach, but was worried about stack allocation. Is there any guarantee that pthread_create will execute the entrypoint before the start function returns?

Reply:
Looks like there aren't guarantees, so the above method won't work.  Is there another way to prevent adding a bunch of stuff to the Task.hpp that other implementors won't use?

Reply:
I think using the class itself is probably necessary to have a static location to store the entrypoint and argument. I think nearly every implementer is going to need a wrapper though, so this isn't adding too much overhead. The only exception is the bare metal scheduler, but it could be structured to use this approach as well.